src/transport/tests/listen-test.js

```diff
@@ -92,6 +92,23 @@ module.exports = (common) => {
       expect(upgradeSpy.callCount).to.equal(2)
     })
 
+    it('should not handle connection if upgradeInbound throws', async () => {
+      sinon.stub(upgrader, 'upgradeInbound').throws()
+
+      const listener = transport.createListener(() => {
+        throw new Error('should not handle the connection if upgradeInbound throws')
+      })
+
+      // Listen
+      await listener.listen(addrs[0])
+
+      // Create a connection to the listener
+      const socket = await transport.dial(addrs[0])
```

Comment:
@vasco-santos I think we should validate the socket was closed as well. Since the listener throws, it should hangup, so the dial socket should get closed. We could wait for `timeline.close` to exist after the dial before closing everything. `await pWaitFor(() => typeof socket.timeline.close === 'number')` or something similar.

Other than that, this looks good. I mainly just want to ensure we are properly closing the sockets on inbound errors.

Reply:
Yeah, I think that is better